src/transport/builder/uattributesbuilder.rs

```diff
@@ -0,0 +1,212 @@
+            priority,
+            ttl: None,
+            token: None,
+            sink: None,
+            plevel: None,
+            commstatus: None,
+            reqid: None,
+        }
+    }
+
+    /// Constructs a `UAttributesBuilder` for a notification message.
+    ///
+    /// # Arguments
+    /// * `priority` - The priority of the message.
+    /// * `sink` - The destination URI.
+    ///
+    /// # Returns
+    /// Returns a `UAttributesBuilder` instance with the configured priority and sink.
+    pub fn notification(priority: UPriority, sink: UUri) -> UAttributesBuilder {
+        UAttributesBuilder {
+            id: UUIDv8Builder::new().build(),
+            message_type: UMessageType::UmessageTypePublish,
+            priority,
+            ttl: None,
+            token: None,
+            sink: Some(sink),
+            plevel: None,
+            commstatus: None,
+            reqid: None,
+        }
+    }
+
+    /// Constructs a `UAttributesBuilder` for a request message.
+    ///
+    /// # Arguments
+    /// * `priority` - The priority of the message.
+    /// * `sink` - The destination URI.
+    /// * `ttl` - The time to live in milliseconds.
+    ///
+    /// # Returns
+    /// Returns a `UAttributesBuilder` instance with the configured priority, sink, and ttl.
+    pub fn request(priority: UPriority, sink: UUri, ttl: i32) -> UAttributesBuilder {
+        UAttributesBuilder {
+            id: UUIDv8Builder::new().build(),
+            message_type: UMessageType::UmessageTypeRequest,
+            priority,
+            ttl: Some(ttl),
+            token: None,
+            sink: Some(sink),
+            plevel: None,
+            commstatus: None,
+            reqid: None,
+        }
+    }
+
+    /// Constructs a `UAttributesBuilder` for a response message.
+    ///
+    /// # Arguments
+    /// * `priority` - The priority of the message.
+    /// * `sink` - The destination URI.
+    /// * `reqid` - The original request UUID used to correlate the response to the request.
+    ///
+    /// # Returns
+    /// Returns a `UAttributesBuilder` instance with the configured priority, sink, and reqid.
+    pub fn response(priority: UPriority, sink: UUri, reqid: Uuid) -> UAttributesBuilder {
+        UAttributesBuilder {
+            id: UUIDv8Builder::new().build(),
+            message_type: UMessageType::UmessageTypeResponse,
+            priority,
+            ttl: None,
+            token: None,
+            sink: Some(sink),
+            plevel: None,
+            commstatus: None,
+            reqid: Some(reqid),
+        }
+    }
+
+    /// Adds the time to live in milliseconds.
+    ///
+    /// # Arguments
+    /// * `ttl` - The time to live in milliseconds.
+    ///
+    /// # Returns
+    /// Returns the `UAttributesBuilder` instance with the configured ttl.
+    pub fn with_ttl(&mut self, ttl: u32) -> &mut UAttributesBuilder {
+        self.ttl = Some(ttl as i32);
```

Comment:
Here we could have an overflow if `ttl` is larger than `i32::MAX`. Should panic? Should return an error?

Reply:
yeah, the joys of vague type definitions... I made an attempt to improve this.